packages/gatsby/src/bootstrap/load-themes/index.ts

```diff
@@ -59,12 +73,12 @@ const resolveTheme = async (
     themeDir,
     `gatsby-config`
   )
-  const theme = preferDefault(configModule)
+  const theme = preferDefault(configModule) as GatsbyConfig
 
   // if theme is a function, call it with the themeConfig
-  let themeConfig = theme
+  let themeConfig: GatsbyConfig = theme
```

Comment:
So `theme` can actually be a function here, which would give `themeConfig` the wrong type. Of course, in the end, it will be of type `GatsbyConfig`. Should I type this in another way? I can also change the runtime behavior to use a ternary.

Reply:
Improved this with @ascorbic suggestion, using a ternary and improving the `theme` type.